pkg/apis/hive/v1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -311,6 +313,19 @@ func validateSecrets(secrets []hivev1.SecretMapping, fldPath *field.Path) field.
 	return allErrs
 }
 
+func validateSourceSecretInSyncSetNamespace(secrets []hivev1.SecretMapping, syncSetNS string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, secret := range secrets {
+		if secret.SourceRef.Namespace != syncSetNS {
```

Comment:
This should also allow an empty namespace, which implies the same namespace as the syncset.

Reply:
Done and added unit test.